docs/adr/0001-create-research-template-docker-repo.md

```diff
@@ -0,0 +1,25 @@
+# 1. Create the research-template-docker repo
+
+Date: 2024-06-04
+
+## Status
+
+Accepted
+
+## Context
+
+We are using [dev containers](https://code.visualstudio.com/docs/devcontainers/create-dev-container) to provide a configuration for Codespaces that gives researchers a useful development environment to use while working on their OpenSAFELY projects. We have created a custom Docker image for this dev container environment. This custom image requires a Dockerfile and other supporting files to be created.
+
+## Decision
+
+We will create a separate GitHub repo to store the Dockerfile and other supporting files. Specifically, we will store the following in the research-template-docker repo:
+
+* The Dockerfile used to generate the dev containers Docker image
+* The tooling and GitHub actions to build the Docker image and to run the tests
+* Some of the supporting files for the research-template (e.g documentation and ADRs).
+
+The research-template repo will only contain the code and configuration to be copied into a researcher’s repo.
+
+## Consequences
+
+By storing the supporting infrastructure in a separate repo we will be creating a good separation of concerns between the research code that will be used to help researchers get started on OpenSAFELY and the supporting infrastructure. The researchers who base their work on the research-template repo will have a copy of the things they need to get started and won't have any of the additional supporting infrastructure. They will not even need to be aware that the supporting infrastructure exists. That supporting infrastructure will be controlled and updated by developers working in the opensafely-core GitHub organisation. Storing the repo in a separate organisation also allows us to apply different security settings to the repo and to more easily setup Dependabot, etc.
```

Comment:
Nygard states that this section should describe the context after the decision has been implemented. There are reasons why locating the configuration in two repositories is a good decision, a neutral decision, and a bad decision. I can't see any reasons why it is a neutral or a bad decision, here. Surely this isn't the best of all possible worlds?

If we had the choice, then we probably wouldn't locate _any_ configuration for the standard development environment alongside the OpenSAFELY project code. However, we don't have the choice.

Reply:
I've tried to reword this to make it a bit clearer and to address your other comment. Please let me know if that's helped.